### PR TITLE
refactor: rename tests (CMC-502)

### PR DESCRIPTION
**Pull request by sabahirfan**
### JIRA link (if applicable) ###
[CMC-502](https://tools.hmcts.net/jira/browse/CMC-502)

### Change description ###
This PR renames the tests and introduces the @Nested classes to group related tests.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/unspec/model/ClaimValueTest.java [hunk 1]
```diff
@@ -1,67 +1,77 @@
 package uk.gov.hmcts.reform.unspec.model;
 
+import org.junit.jupiter.api.Nested;
 import org.junit.jupiter.api.Test;
 
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
+@SuppressWarnings("HideUtilityClassConstructor")
```
Comment (sabahirfan) [inline src/test/java/uk/gov/hmcts/reform/unspec/model/ClaimValueTest.java:11]: Started complaining after introducing `@Nested`
 ```
ClaimValueTest.java:9:1: Utility classes should not have a public or default constructor. [HideUtilityClassConstructor]
```

File: src/test/java/uk/gov/hmcts/reform/unspec/helpers/DateFormatHelperTest.java [hunk 2]
```diff
@@ -13,19 +14,25 @@
...
+                .isEqualTo("9:00am on 1 January 2999");
+        }
     }
 
-    @Test
-    void shouldFormatDateToExpectedFormat() {
-        LocalDate now = LocalDate.of(2999, 1, 1);
+    @Nested
+    class LocalDateFormat {
+        @Test
```
Comment (swalker125) [inline src/test/java/uk/gov/hmcts/reform/unspec/helpers/DateFormatHelperTest.java:32]: think no space here should fail checkstyle now?

File: src/test/java/uk/gov/hmcts/reform/service/UserServiceTest.java [hunk 3]
```diff
@@ -49,7 +49,7 @@ public void setup() {
     }
 
     @Test
-    void findsUserInfoForAuthToken() {
+    void shouldReturnUserInfo_whenValidAuthToken() {
```
Comment (pd-dudkiewicz) [inline src/test/java/uk/gov/hmcts/reform/service/UserServiceTest.java]: can we remove the space here?

File: src/test/java/uk/gov/hmcts/reform/unspec/advice/ResourceExceptionHandlerTest.java [hunk 4]
```diff
@@ -20,7 +20,7 @@ public void setUp() {
     }
 
     @Test
-    public void shouldReturnNotFoundWhenCallbackException() {
+    public void shouldReturnNotFound_whenCallbackException() {
```
Comment (pd-dudkiewicz) [inline src/test/java/uk/gov/hmcts/reform/unspec/advice/ResourceExceptionHandlerTest.java]: should it be `whenCallbackExceptionThrown`?

File: src/test/java/uk/gov/hmcts/reform/unspec/callback/CallbackHandlerFactoryTest.java [hunk 5]
```diff
@@ -70,7 +70,7 @@ public void shouldThrowCallbackExceptionWhenUnknownEvent() {
     }
 
     @Test
-    public void shouldDispatchCallbackWhenCreateCaseEvent() {
+    public void shouldDispatchCallback_whenValidCaseEvent() {
```
Comment (pd-dudkiewicz) [inline src/test/java/uk/gov/hmcts/reform/unspec/callback/CallbackHandlerFactoryTest.java]: one more empty line

File: src/test/java/uk/gov/hmcts/reform/unspec/callback/CallbackTypeTest.java [hunk 6]
```diff
@@ -8,7 +8,7 @@
 public class CallbackTypeTest {
 
     @Test
-    public void shouldDeserialiseWhenValidCallbacks() {
+    public void shouldDeserialise_whenValidCallbacks() {
```
Comment (pd-dudkiewicz) [inline src/test/java/uk/gov/hmcts/reform/unspec/callback/CallbackTypeTest.java]: deserialiSe or deserialiZe? 🤔 it's marked as typo for me now

File: src/test/java/uk/gov/hmcts/reform/unspec/helpers/DateFormatHelperTest.java (hunk 2 above)
Comment (sabahirfan) [reply src/test/java/uk/gov/hmcts/reform/unspec/helpers/DateFormatHelperTest.java:32]: yes should have. will look into that separately.

File: src/test/java/uk/gov/hmcts/reform/unspec/enums/ServiceMethodTest.java [hunk 7]
```diff
@@ -12,57 +13,69 @@
...
 
-        assertThat(serviceMethod.getDeemedDateOfService(dateTime.atStartOfDay()))
-            .isEqualTo(dateTime.plusDays(serviceMethod.getDays()));
-    }
+        @ParameterizedTest
+        @EnumSource(value = ServiceMethod.class, names = {"POST", "DOCUMENT_EXCHANGE", "FAX", "EMAIL", "OTHER"})
+        void shouldReturnExpectedDays_whenTimeIsBefore4pm(ServiceMethod serviceMethod) {
+            LocalDate dateTime = LocalDate.of(2000, 1, 1);
+
+            assertThat(serviceMethod.getDeemedDateOfService(dateTime.atStartOfDay()))
```
Comment (pd-dudkiewicz) [inline src/test/java/uk/gov/hmcts/reform/unspec/enums/ServiceMethodTest.java:24]: Should we separate [Arrange-Act-Assert](https://github.com/testdouble/contributing-tests/wiki/Arrange-Act-Assert) steps? It's not part of the task, so happy to leave it as it is for now.

File: src/test/java/uk/gov/hmcts/reform/unspec/handler/ConfirmServiceCallbackHandlerTest.java [hunk 8]
```diff
@@ -24,73 +25,95 @@
...
+            assertThat(response.getErrors()).containsExactly(
+                "CONTENT TBC: please enter a valid value for other documents");
+        }
 
-    @Test
-    void shouldReturnNoErrorWhenValidServedDocumentsOtherValue() {
-        Map<String, Object> data = new HashMap<>();
-        data.put("servedDocumentsOther", "A valid document");
+        @Test
+        void shouldReturnNoError_whenValidServedDocumentsOtherValue() {
```
Comment (pd-dudkiewicz) [inline src/test/java/uk/gov/hmcts/reform/unspec/handler/ConfirmServiceCallbackHandlerTest.java]: I'd remove `Value` bit

File: src/test/java/uk/gov/hmcts/reform/unspec/handler/ConfirmServiceCallbackHandlerTest.java [hunk 9]
```diff
@@ -24,73 +25,95 @@
...
-    void shouldReturnExpectedAboutToSubmitCallbackResponseObject() {
-        Map<String, Object> data = new HashMap<>();
-        data.put("serviceMethod", "POST");
-        data.put("serviceDate", "2099-06-23");
+    @Nested
+    class AboutToSubmitCallback {
 
-        CallbackParams params = callbackParamsOf(data, CallbackType.ABOUT_TO_SUBMIT);
+        @Test
+        void shouldReturnExpectedAboutToSubmitCallbackResponse_whenNoError() {
```
Comment (pd-dudkiewicz) [inline src/test/java/uk/gov/hmcts/reform/unspec/handler/ConfirmServiceCallbackHandlerTest.java]: How about `whenValidData`?

File: src/test/java/uk/gov/hmcts/reform/unspec/handler/ConfirmServiceCallbackHandlerTest.java [hunk 10]
```diff
@@ -24,73 +25,95 @@
...
 
-    @Test
-    void shouldReturnExpectedSubmittedCallbackResponseObject() {
-        Map<String, Object> data = new HashMap<>();
-        data.put("deemedDateOfService", "2099-06-25");
+    @Nested
+    class SubmittedCallback {
+
+        @Test
+        void shouldReturnExpectedSubmittedCallbackResponse_whenNoError() {
```
Comment (pd-dudkiewicz) [inline src/test/java/uk/gov/hmcts/reform/unspec/handler/ConfirmServiceCallbackHandlerTest.java]: As mentioned above

**Review by dharmendrak — DISMISSED** [review]

File: src/test/java/uk/gov/hmcts/reform/unspec/handler/CreateClaimCallbackHandlerTest.java [hunk 11]
```diff
@@ -32,79 +33,98 @@
...
-
-        AboutToStartOrSubmitCallbackResponse response = (AboutToStartOrSubmitCallbackResponse) handler.handle(params);
-
-        assertThat(response.getErrors())
-            .containsOnly("CONTENT TBC: Higher value must not be lower than the lower value.");
+    @Nested
+    class MidEventCallback {
+
+        @Test
+        void shouldReturnExpectedErrorInMidEvent_whenValuesAreInvalid() {
```
Comment (pd-dudkiewicz) [inline src/test/java/uk/gov/hmcts/reform/unspec/handler/CreateClaimCallbackHandlerTest.java]: I think we could remove `InMidEvent` part as it's there in nested class name

File: src/test/java/uk/gov/hmcts/reform/unspec/handler/CreateClaimCallbackHandlerTest.java [hunk 12]
```diff
@@ -32,79 +33,98 @@
...
+
+            AboutToStartOrSubmitCallbackResponse response = (AboutToStartOrSubmitCallbackResponse) handler
+                .handle(params);
+
+            assertThat(response.getErrors())
+                .containsOnly("CONTENT TBC: Higher value must not be lower than the lower value.");
+        }
+
+        @Test
+        void shouldReturnNoErrorInMidEvent_whenValuesAreValid() {
```
Comment (pd-dudkiewicz) [inline src/test/java/uk/gov/hmcts/reform/unspec/handler/CreateClaimCallbackHandlerTest.java]: same here

File: src/test/java/uk/gov/hmcts/reform/unspec/handler/CreateClaimCallbackHandlerTest.java [hunk 13]
```diff
@@ -32,79 +33,98 @@
...
-            ));
-    }
+    @Nested
+    class AboutToSubmitCallback {
 
-    @Test
-    void shouldAddClaimIssuedDateInAboutToSubmit() {
-        CallbackParams params = callbackParamsOf(new HashMap<>(), CallbackType.ABOUT_TO_SUBMIT);
+        @Test
+        void shouldAddClaimIssuedDate_whenAboutToSubmitInvoked() {
```
Comment (pd-dudkiewicz) [inline src/test/java/uk/gov/hmcts/reform/unspec/handler/CreateClaimCallbackHandlerTest.java]: maybe `whenInvoked`? We've got callback type is in class name

File: src/test/java/uk/gov/hmcts/reform/unspec/handler/CreateClaimCallbackHandlerTest.java [hunk 14]
```diff
@@ -32,79 +33,98 @@
...
-        assertThat(response).isEqualToComparingFieldByField(
-            SubmittedCallbackResponse.builder()
-                .confirmationHeader("# Your claim has been issued\n## Claim number: TBC")
-                .confirmationBody(body)
-                .build());
+    @Nested
+    class SubmittedCallback {
+
+        @Test
+        void shouldReturnExpectedSubmittedCallbackResponse_whenSubmitted() {
```
Comment (pd-dudkiewicz) [inline src/test/java/uk/gov/hmcts/reform/unspec/handler/CreateClaimCallbackHandlerTest.java]: Maybe we should be more specific here? And not mention callback name as mentioned above, for example something like `shouldReturnConfirmationData_whenInvoked`?

**Review by pd-dudkiewicz — COMMENTED** [review]
Left a few comments. Also I think we didn't agree on how we should name nested classes and what exactly should be their purpose. Was it about grouping tests logically or based on the method they test. Happy to go either way as long as we keep it consistent.

File: src/test/java/uk/gov/hmcts/reform/unspec/callback/CallbackHandlerFactoryTest.java (hunk 5 above)
Comment (sabahirfan) [reply src/test/java/uk/gov/hmcts/reform/unspec/callback/CallbackHandlerFactoryTest.java]: removed

File: src/test/java/uk/gov/hmcts/reform/service/UserServiceTest.java (hunk 3 above)
Comment (sabahirfan) [reply src/test/java/uk/gov/hmcts/reform/service/UserServiceTest.java]: removed

File: src/test/java/uk/gov/hmcts/reform/unspec/callback/CallbackTypeTest.java (hunk 6 above)
Comment (sabahirfan) [reply src/test/java/uk/gov/hmcts/reform/unspec/callback/CallbackTypeTest.java]: changed to deserialiZe

File: src/test/java/uk/gov/hmcts/reform/unspec/enums/ServiceMethodTest.java (hunk 7 above)
Comment (sabahirfan) [reply src/test/java/uk/gov/hmcts/reform/unspec/enums/ServiceMethodTest.java:24]: agree that is not in scope for this task.

File: src/test/java/uk/gov/hmcts/reform/unspec/handler/ConfirmServiceCallbackHandlerTest.java [hunk 15]
```diff
@@ -24,73 +25,95 @@
...
-    @Test
-    void shouldPrepopulateServedDocumentsList() {
-        CallbackParams params = callbackParamsOf(new HashMap<>(), CallbackType.ABOUT_TO_START);
-        AboutToStartOrSubmitCallbackResponse response = (AboutToStartOrSubmitCallbackResponse) handler.handle(params);
+    @Nested
+    class AboutToStartCallback {
 
-        assertThat(response.getData()).isEqualTo(Map.of("servedDocuments", List.of(ServedDocuments.CLAIM_FORM)));
+        @Test
+        void shouldPrepopulateServedDocumentsList_whenAboutToStartEvent() {
```
Comment (pd-dudkiewicz) [inline src/test/java/uk/gov/hmcts/reform/unspec/handler/ConfirmServiceCallbackHandlerTest.java]: should be `_whenInvoked`

File: src/test/java/uk/gov/hmcts/reform/unspec/handler/ConfirmServiceCallbackHandlerTest.java [hunk 16]
```diff
@@ -24,73 +25,95 @@
...
-    void shouldReturnExpectedAboutToSubmitCallbackResponseObject() {
-        Map<String, Object> data = new HashMap<>();
-        data.put("serviceMethod", "POST");
-        data.put("serviceDate", "2099-06-23");
+    @Nested
+    class AboutToSubmitCallback {
 
-        CallbackParams params = callbackParamsOf(data, CallbackType.ABOUT_TO_SUBMIT);
+        @Test
+        void shouldReturnExpectedAboutToSubmitCallbackResponse_whenValidData() {
```
Comment (pd-dudkiewicz) [inline src/test/java/uk/gov/hmcts/reform/unspec/handler/ConfirmServiceCallbackHandlerTest.java]: `shouldReturnExpectedResponse_whenValidData`?

File: src/test/java/uk/gov/hmcts/reform/unspec/handler/ConfirmServiceCallbackHandlerTest.java [hunk 17]
```diff
@@ -24,73 +25,95 @@
...
 
-    @Test
-    void shouldReturnExpectedSubmittedCallbackResponseObject() {
-        Map<String, Object> data = new HashMap<>();
-        data.put("deemedDateOfService", "2099-06-25");
+    @Nested
+    class SubmittedCallback {
+
+        @Test
+        void shouldReturnExpectedSubmittedCallbackResponse_whenValidData() {
```
Comment (pd-dudkiewicz) [inline src/test/java/uk/gov/hmcts/reform/unspec/handler/ConfirmServiceCallbackHandlerTest.java]: as above, `shouldReturnExpectedResponse_whenValidData`?

File: src/test/java/uk/gov/hmcts/reform/unspec/handler/CreateClaimCallbackHandlerTest.java [hunk 18]
```diff
@@ -32,79 +33,98 @@
...
-        assertThat(response).isEqualToComparingFieldByField(
-            SubmittedCallbackResponse.builder()
-                .confirmationHeader("# Your claim has been issued\n## Claim number: TBC")
-                .confirmationBody(body)
-                .build());
+    @Nested
+    class SubmittedCallback {
+
+        @Test
+        void shouldReturnExpectedSubmittedCallbackResponse_whenInvoked() {
```
Comment (pd-dudkiewicz) [inline src/test/java/uk/gov/hmcts/reform/unspec/handler/CreateClaimCallbackHandlerTest.java]: `shouldReturnExpectedResponse_whenInvoked`

**Review by pd-dudkiewicz — APPROVED** [review]

**Review by dharmendrak — APPROVED** [review]

**Pull request merged**